atlasdb-jdbc/src/main/java/com/palantir/atlasdb/keyvalue/jdbc/JdbcKeyValueConfiguration.java

```diff
@@ -43,6 +43,11 @@ public String getTablePrefix() {
         return "at_";
     }
 
+    @Value.Default
+    public int getBatchSizeForReads() {
+        return 10_000;
```

Comment:
I would add a comment noting that this isn't arbitrary, it specifically needs to be approximately < (32_000 / 3).  Or even a check since we know it will fail with anything higher.